state/state.go

```diff
@@ -448,6 +441,47 @@ func (s *State) CloseBatch(ctx context.Context, receipt ProcessingReceipt, dbTx
 	if isLastBatchClosed {
 		return ErrBatchAlreadyClosed
 	}
+
+	return nil
+}
+
+// CloseSynchronizedBatch is used by Synchronizer to close the current batch.
+func (s *State) CloseSynchronizedBatch(ctx context.Context, receipt ProcessingReceipt, txs []types.Transaction, dbTx pgx.Tx) error {
```

Comment:
I understand that this function should be call only by `ProcessAndStoreClosedBatch`. If this is the case, please make the function private.

Please keep in mind that the synchronizer also calls the Open / Close batch methods, when synchronizing from the trusted sequencer broadcast. In this scenario, whoever is implementing this could be very tempted to call `CloseSynchronizedBatch` instead of `CloseBatch`